src/VisualStudio/Core/Def/InheritanceMargin/InheritanceMarginTaggerProvider.cs

```diff
@@ -65,14 +65,9 @@ protected override bool CanCreateTagger(ITextView textView, ITextBuffer buffer)
 
     protected override ITaggerEventSource CreateEventSource(ITextView textView, ITextBuffer subjectBuffer)
     {
-        // Because we use frozen-partial documents for semantic classification, we may end up with incomplete
-        // semantics (esp. during solution load).  Because of this, we also register to hear when the full
-        // compilation is available so that reclassify and bring ourselves up to date.
         // Note: Also generate tags when InheritanceMarginOptions.InheritanceMarginCombinedWithIndicatorMargin is changed,
         // because we want to refresh the glyphs in indicator margin.
-        return new CompilationAvailableTaggerEventSource(
-           subjectBuffer,
-           AsyncListener,
+        return TaggerEventSources.Compose(
```

Comment:
>return TaggerEventSources.Compose( [](http://example.com/codeflow?start=8&length=34)

There's still a comment reference to CompilationAvailableTaggerEventSource below

Reply:
good to know!